pkg/infrastructure/azure/azure.go

```diff
@@ -339,3 +344,49 @@ func randomString(length int) string {
 
 	return string(s)
 }
+
+// Ignition provisions the Azure container that holds the bootstrap ignition
+// file.
+func (p Provider) Ignition(ctx context.Context, in clusterapi.IgnitionInput) ([]byte, error) {
+	session, err := in.InstallConfig.Azure.Session()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get session: %w", err)
+	}
+
+	bootstrapIgnData := in.BootstrapIgnData
+	subscriptionID := session.Credentials.SubscriptionID
+	cloudConfiguration := session.CloudConfig
+
+	ignitionContainerName := "ignition"
+	blobName := "bootstrap.ign"
+	blobURL := fmt.Sprintf("%s/%s/%s", p.StorageURL, ignitionContainerName, blobName)
+
+	// Create ignition blob storage container
+	createBlobContainerOutput, err := CreateBlobContainer(ctx, &CreateBlobContainerInput{
+		ContainerName:        ignitionContainerName,
+		SubscriptionID:       subscriptionID,
+		ResourceGroupName:    p.ResourceGroupName,
+		StorageAccountName:   p.StorageAccountName,
+		StorageClientFactory: p.StorageClientFactory,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	blobIgnitionContainer := createBlobContainerOutput.BlobContainer
+	logrus.Debugf("BlobIgnitionContainer.ID=%s", *blobIgnitionContainer.ID)
+
+	_, err = CreateBlockBlob(ctx, &CreateBlockBlobInput{
+		StorageURL:         p.StorageURL,
+		BlobURL:            blobURL,
+		StorageAccountName: p.StorageAccountName,
+		StorageAccountKeys: p.StorageAccountKeys,
+		CloudConfiguration: cloudConfiguration,
+		BootstrapIgnData:   bootstrapIgnData,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return []byte{}, nil
```

Comment:
this `[]byte{}` needs to include the ignition shim, which should include a redirect to the url for the sas link to the ignition bucket. @rna-afk has a good start on creating the ignition shim in #8155